File: day:71/findMedianSortedArrays.py
from math import floor

def merge_sorted_arrays(nums1, nums2):
    # * --- Brute force method - ------------------------ * /
    result = []
    [i, j] = [0, 0]

    while i < len(nums1) and j < len(nums2):
        if nums1[i] <= nums2[j]:
            result.append(nums1[i])
            i += 1
        else:
            result.append(nums2[j])
            j += 1
    if i < len(nums1):
        for jj in range(i, len(nums1)):
            result.append(nums1[jj])
    if j < len(nums2):
        for ii in range(j, len(nums2)):
            result.append(nums2[ii])
    return result


def median(num):
    # odd length
    if len(num) % 2 == 1:
        return num[floor(len(num) / 2)]
    else:
        # even length
        mid = floor(len(num) / 2)
        return (num[mid - 1] + num[mid]) / 2
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        result = merge_sorted_arrays(nums1, nums2)
        return median(result);
        

File: day:71/findMedianSortedArrays.java
class Solution {
    public int[] merge_sorted_arrays(int[] nums1, int[] nums2) {
        /* --- Brute force method ------------------------- */
        int[] result = new int[nums1.length + nums2.length];
        int i, j , k;
        i = j = k = 0;

        while(i < nums1.length && j < nums2.length) {
            if(nums1[i] <= nums2[j]) {
                result[k] = nums1[i];
                i += 1;
            }
            else {
                result[k] = nums2[j];
                j += 1;
            }
            k += 1;
        }
        if(i < nums1.length) {
            for(int jj = i; jj < nums1.length; jj++) {
                result[k] = nums1[jj];
                k += 1;
            }
        }
        if(j < nums2.length) {
            for(int ii = j; ii < nums2.length; ii++) {
                result[k] = nums2[ii];
                k += 1;
            }
        }

        return result;
    }
    public double median(int[] num) {
        // odd length
        if(num.length % 2 == 1) {
            return num[(int)Math.floor(num.length/2)];
        }
        else {
            // even length
            int mid = (int) Math.floor(num.length/2);
            return (num[mid - 1] + num[mid]) /2.0 ;
        }
    }
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int[] result = merge_sorted_arrays(nums1, nums2);
        return median(result);
    }
}
